typescript compilation errors with nodejs 16

The integration tests in the estuary/connectors repo are failing with typescript compilation errors like:

```
> catalog-js-transformer@0.0.0 compile /home/flow/project/.build/tests/source-kinesis
> tsc

node_modules/@types/node/child_process.d.ts:267:18 - error TS2304: Cannot find name 'AbortSignal'.

267         signal?: AbortSignal | undefined;
                     ~~~~~~~~~~~
```

Newer versions of nodejs include the `AbortController` API. There were Typescript definitions added to the `@node/types` package that reference these `Abort_` types, but `tsc` seems to have no idea how to actually resolve those types.
According to [this issue](https://stackoverflow.com/a/59893096), this can be worked around by adding `DOM` to the tsc `lib` config.

Additionally, while investigating this issue, I found that the nodejs version was unintentionally being upgraded while building the docker image, and the current flow image is shipping nodejs 16, not 14. I'm not sure whether it's worth "fixing" this to pin to v14, or if we should just make the version 16 explicit.

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -5,7 +5,12 @@
 		"forceConsistentCasingInFileNames": true,
 		"incremental": true,
 		"lib": [
-			"ES2019"
+			"ES2019",
+			// DOM is added here only to provide the `AbortController` types, which are
+			// used by other nodejs types. This fix was taken from
+			// (https://stackoverflow.com/a/59893096) to fix the Typescript compilation
+			// issue described in estuary/flow#279
+			"DOM"
 		],
 		"module": "commonjs",
 		"noEmitOnError": true,

